scripts/deploy.py

```diff
@@ -43,18 +43,68 @@ def upload_archive(archive, zenodo_json, version):
     headers = {"Accept": "application/json"}
     params = {"access_token": ZENODO_TOKEN}
 
-    # Create an empty upload
-    response = requests.post(
-        "https://zenodo.org/api/deposit/depositions",
-        params=params,
-        json={},
-        headers=headers,
-    )
-    if response.status_code not in [200, 201]:
-        sys.exit(
-            "Trouble requesting new upload: %s, %s"
-            % (response.status_code, response.json())
+    if doi:
+        depositions = requests.get(
+            "https://zenodo.org/api/deposit/depositions",
+            params=params,
+            headers=headers,
+        )
+        if depositions.status_code not in [200, 201]:
+            sys.exit("Cannot query depositions: %s, %s" % (depositions.status_code, depositions.json()))
+
+        target_deposit = None
+        for deposit in depositions.json():
+            print("looking at deposit %s" % (deposit['doi']))
```

Comment:
One debug log I found helpful. I don't know how it'd affect other programs that wouldn't expect `stdout`? Just calling it out because I don't know if we should be more careful about stdout.

Reply:
For a GitHub action we can be fairly confident it's going to run there - I'd expect people that want to run stuffs locally to use a python wrapper for the zenodo API.